lib/mocha/mockery.rb

```diff
@@ -7,16 +7,47 @@
 require 'mocha/logger'
 require 'mocha/configuration'
 require 'mocha/stubbing_error'
+require 'mocha/not_initialized_error'
 require 'mocha/expectation_error_factory'
 
 module Mocha
 
   class Mockery
 
+    class Null < self
+
+      def add_mock(*)
+        raise_not_initialized_error
+      end
+
+      def add_state_machine(*)
+        raise_not_initialized_error
+      end
+
+      def stubba
+        Central::Null.new(&method(:raise_not_initialized_error))
+      end
+
+      private
+
+      def raise_not_initialized_error
+        message = 'Mocha methods cannot be used outside the context of a test'
+        raise NotInitializedError.new(message, caller)
+      end
+
+    end
+
     class << self
 
       def instance
-        @instance ||= new
+        @instances.last || Null.new
```

Comment:
This throws `undefined method `last' for nil:NilClass` when you attempt to use mocha uninitialized (e.g. example snippets from documentation).

Reply:
@lzap Gah! Looks like I went wrong with some of my interactive rebasing. Thanks for testing - I'll try to sort it out as soon as I can.

Reply:
@lzap I'm in the process of fixing this and just to warn you I plan to force-push the branch shortly.